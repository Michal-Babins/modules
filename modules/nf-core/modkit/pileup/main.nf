process MODKIT_PILEUP {
    tag "$meta.id"
    label 'process_single'

    conda "bioconda::ont-modkit=0.2.2"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/ont-modkit:0.2.2--hdcf5f25_0':
        'biocontainers/ont-modkit:0.2.2--hdcf5f25_0' }"

    input:
    tuple val(meta), path(bam), val(out_bed)

    output:
    tuple val(meta), path(out_bed), emit: bed
    tuple val(meta), path("*_pileup.log"), emit: log
    path "versions.yml"           , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    
    """
    modkit \\
        pileup \\
        $bam \\
        $out_bed \\
        --threads $task.cpus \\
        --log-filepath ${prefix}_pileup.log \\
        $args


    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        : \$(echo \$(modkit --version 2>&1) | sed 's/^.*modkit //; s/Using.*\$//' ))
    END_VERSIONS
    """

    stub:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    """
    touch $out_bed
    touch ${prefix}_pileup.log

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        : \$(echo \$(samtools --version 2>&1) | sed 's/^.*samtools //; s/Using.*\$//' ))
    END_VERSIONS
    """
}
