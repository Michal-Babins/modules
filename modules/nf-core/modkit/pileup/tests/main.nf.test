nextflow_process {

    name "Test Process MODKIT_PILEUP"
    script "../main.nf"
    process "MODKIT_PILEUP"
    tag "modules"
    tag "modules_nfcore"
    tag "modkit"
    tag "modkit_pileup"

    test("modkit_pileup") {

        when {
            process {
                """
                input[0] = [
                    [ id:'test', single_end:false ],
                    [ file('https://github.com/nf-core/test-datasets/raw/modules/data/genomics/sarscov2/illumina/bam/test.paired_end.sorted.bam', checkIfExists: true) ]
                ]
                input[1] = file('https://github.com/nf-core/test-datasets/raw/modules/data/genomics/sarscov2/illumina/bam/test.paired_end.sorted.bam.bai', checkIfExists: true) 
                input[2] = file("test.bed") 
                """
            }
        }

        then {
            assertAll (
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

}