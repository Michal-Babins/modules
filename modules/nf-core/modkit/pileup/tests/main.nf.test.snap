{
    "modkit_pileup": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.bed:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.paired_end.sorted_pileup.log:md5,218e492be12d7224cd2d3ff78c7a4986"
                    ]
                ],
                "2": [
                    "versions.yml:md5,f93491243c04370eaf7d1053669c46e9"
                ],
                "bed": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.bed:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "log": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.paired_end.sorted_pileup.log:md5,218e492be12d7224cd2d3ff78c7a4986"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,f93491243c04370eaf7d1053669c46e9"
                ]
            }
        ],
        "timestamp": "2023-11-29T06:12:02.440619"
    }
}